pkg/test/framework/components/zipkin/zipkin.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zipkin
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+// Instance represents a zipkin deployment on kube
+type Instance interface {
+	resource.Resource
+
+	// QueryTraces returns all available traces via zipkin query api
+	QueryTraces() ([]Trace, error)
+}
+
+// Span represents a single span, which includes span attributes for verification
+type Span struct {
```

Comment:
Rather than defining our own structures, does Zipkin already define these? I see this: https://github.com/openzipkin/zipkin-go/blob/master/proto/v2/zipkin.proto.

If we define this API as protos, then we can potentially leverage our [structpath library](https://github.com/istio/istio/blob/master/pkg/test/util/structpath/instance.go) to run JSON-path queries against the responses.

Reply:
I tried but dep ensure does not add that proto.pb.go since it is not used anywhere in that repo.

I also tried to feed in the json text from zipkin rest API into the structpath and verify the spans with jsonPath to avoid defining this struct, but it is not as clear comparing to use this tree-like data struct, also seems like it is not as simple, for example I cannot find an easy query to get the length of a struct array.

Reply:
Ok, can you raise a bug to further investigate switching this API over to use the zipkin proto?

Then add a TODO here that references the bug.